status/reporter_github.go

```diff
@@ -86,37 +87,46 @@ func NewCheckRunStatusUpdater(
 }
 
 func GetAppCredentials(ctx context.Context, k8sclient client.Client, object client.Object) (*appCredentials, error) {
-	var err error
+	log := log.FromContext(ctx)
+	var err, unRecoverableError error
 	var found bool
 	appInfo := appCredentials{}
 
 	appInfo.InstallationID, err = strconv.ParseInt(object.GetAnnotations()[gitops.PipelineAsCodeInstallationIDAnnotation], 10, 64)
 	if err != nil {
-		return nil, err
+		unRecoverableError = helpers.NewUnrecoverableMetadataError(fmt.Sprintf("Error %s when parsing string annotation %s: %s", err.Error(), gitops.PipelineAsCodeInstallationIDAnnotation, object.GetAnnotations()[gitops.PipelineAsCodeInstallationIDAnnotation]))
+		log.Error(unRecoverableError, fmt.Sprintf("Error %s when parsing string annotation %s: %s", err.Error(), gitops.PipelineAsCodeInstallationIDAnnotation, object.GetAnnotations()[gitops.PipelineAsCodeInstallationIDAnnotation]))
+		return nil, unRecoverableError
 	}
 
 	// Get the global pipelines as code secret
 	pacSecret := v1.Secret{}
 	err = k8sclient.Get(ctx, types.NamespacedName{Namespace: integrationNS, Name: PACSecret}, &pacSecret)
 	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to get pac secret %s/%s", integrationNS, PACSecret))
 		return nil, err
 	}
 
 	// Get the App ID from the secret
 	ghAppIDBytes, found := pacSecret.Data[gitHubApplicationID]
 	if !found {
-		return nil, errors.New("failed to find github-application-id secret key")
+		unRecoverableError = helpers.NewUnrecoverableMetadataError("failed to find github-application-id secret key")
+		log.Error(unRecoverableError, "failed to find github-application-id secret key")
+		return nil, unRecoverableError
 	}
 
 	appInfo.AppID, err = strconv.ParseInt(string(ghAppIDBytes), 10, 64)
 	if err != nil {
-		return nil, err
+		unRecoverableError = helpers.NewUnrecoverableMetadataError(fmt.Sprintf("Error %s when parsing ghAppIDBytes", err.Error()))
+		log.Error(unRecoverableError, "Error when parsing ghAppIDBytes")
```

Comment:
suggestion: maybe using something like `Error when parsing gitHub App ID` would be more clear to the users?

Reply:
Updated. Thanks!